PEAR-2030: Fix for Cohort Builder, number ranges, difficult to type in a number

paribartandhakal opened the pull request:
## Description
Includes refactor for `NumericRangeFacet.tsx`

_**PLEASE VERIFY ALL THE NUMERIC RANGE FACETS IF POSSIBLE**_ 🙏 

## Checklist
- [X] Added proper unit tests
- [N/A] Left proper TODO messages for any remaining tasks
- [N/A] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

## Screenshots/Screen Recordings (if Appropriate)
![1](https://github.com/user-attachments/assets/3e69eb39-3852-41d7-bad1-ecd6e360bfd7)
![2](https://github.com/user-attachments/assets/66c0dd79-df5d-4862-8782-10723b4e616f)
![3](https://github.com/user-attachments/assets/74106ba3-7170-44ea-9b42-c899fdc166a3)

Commit:
initial setup

Commit:
update useEffect

Commit:
use useForm

Commit:
change the file name

wteouchicago commented:
1. Minor The word "range" is a bit weird for these, esp now that we have added "Min" and "Max". Can we drop it? I think "Min: 0" / "Max: 999999" should convey whatever we need. cc @ebusygina 

<img width="231" alt="Screenshot 2024-08-05 at 1 00 34 PM" src="https://github.com/user-attachments/assets/4e02cbb9-7a8a-4589-878d-8e8c276d52f0">

__
Looks fine otherwise. 👍 Evgeniia can test more once we are on qa.

Commit:
make units optional for fromto

paribartandhakal commented:
> 1. Minor The word "range" is a bit weird for these, esp now that we have added "Min" and "Max". Can we drop it? I think "Min: 0" / "Max: 999999" should convey whatever we need. cc @ebusygina
> 
> <img alt="Screenshot 2024-08-05 at 1 00 34 PM" width="231" src="https://private-user-images.githubusercontent.com/73256434/355198045-4e02cbb9-7a8a-4589-878d-8e8c276d52f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4ODI5ODIsIm5iZiI6MTcyMjg4MjY4MiwicGF0aCI6Ii83MzI1NjQzNC8zNTUxOTgwNDUtNGUwMmNiYjktN2E4YS00NTg5LTg3OGQtOGU4YzI3NmQ1MmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA1VDE4MzEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmNTBlNGIyZjBkMTY2ZTU1ODE4ODhhMjVlZDNhZWU2MGY1MDA1ZjFlMWQ3YTI2ZWZlNjAyNmJhNGMyM2UwZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fwJmBk5w4WaZjkYVuEPnHvOYH8WVVcMVXvfh_ee4l70">
> __ Looks fine otherwise. 👍 Evgeniia can test more once we are on qa.

Just updated it.

wteouchicago commented:
looks good. 👍

craigrbarnes commented on packages/portal-proto/src/features/facets/NumericRangeFacet/FromTo.tsx:
```diff
@@ -0,0 +1,309 @@
+}: WarningOrErrorProps) => (
+  <div className="flex flex-col gap-1 my-1">
+    {hasErrors ? (
+      <span className="text-[#AD2B4A]">
```
We define the tailwind color "utility-error" which would be better to use.

paribartandhakal replied:
I updated this.

craigrbarnes commented:
Also, it is possible to vertically align the icon and the message 
The icon seems too small to me; I find it hard to see the **!**  (JC might think so too).

Commit:
update classname, layout

paribartandhakal commented:
> Also, it is possible to vertically align the icon and the message The icon seems too small to me; I find it hard to see the **!** (JC might think so too).

I have made the changes but waiting for @cemile-ctds' verification.

Commit:
update style for from to

craigrbarnes reviewed (approved):
The refactoring is quite extensive, but overall, it's fine.
